src/content/docs/new-relic-solutions/best-practices-guides/full-stack-observability/synthetic-monitoring-best-practices-guide.mdx

```diff
@@ -267,6 +267,96 @@ console.log("Duration: " + script_duration)
 
     </Step>
 
+    <Step>
+    ## Configure and develop a scripted API test [#scripted-api-test]
+    Using scripted API monitors, you can build complex monitoring workflows with Node.js and the `got` module. For example, you can authenticate with an API, handle errors based on the response code, and parse the response body. To do this, you'll:
+
+    1. Go to <DoNotTranslate>**[one.newrelic.com > All capabilities](https://one.newrelic.com/all-capabilities) > Synthetic Monitoring**</DoNotTranslate>.
+    2. Click the <DoNotTranslate>**Create monitor**</DoNotTranslate> button.
+    3. Enter the name and details of your monitor. For example, `sitename.com`.
+    4. Select the locations from where you want your monitor to run. For example, Mumbai, Seoul, Columbus, or Montreal.
+    5. Choose a frequency for how often your monitor will run on each location selected. For example, `5 mins`.
+    6. Set a notification method to alert your team when performance incidents occur.
+    7. You are now ready to write your script. See this example script to test making an API request and processing the response.
+
+```js
+/**
+ * Script Name: Best Practices - Node 16.10.0
+ * Author:      New Relic
+ * Version:     1.0
+ * Purpose:     Best practices example of a New Relic Synthetics scripted API monitor with async/await, secure credentials for authentication,
+ *              assert on statusCode, retries, error handling, json parsing, $util.insights, and a custom header.
+ * Reference:   https://docs.newrelic.com/docs/synthetics/synthetic-monitoring/scripting-monitors/write-synthetic-api-tests/
+ */
+
+const MAX_RETRIES = 3
+const RETRY_DELAY_MS = 200 // Delay of 200 milliseconds between retries
+
+var assert = require("assert")
+
+const makeGetRequest = async (url, options, maxRetries) => {
+  for (let attempt = 1; attempt <= maxRetries; attempt++) {
+    try {
+      console.log(`Attempt ${attempt} of ${maxRetries}`)
```

Comment:
Do we need to encode retries when the monitor will retry 3 times anyway? If the monitor is having issues, up to 9 attempts could be made.

Reply:
I've seen cases where monitors failed because the script was not robust enough to handle a little latency from the New Relic side. If they had retried after some delay, it would have been a different outcome.

Reply:
Advanced example now utilizes `got` module retries for improved reliability.